gk/main.c

```diff
@@ -654,6 +759,13 @@ cleanup_gk(struct gk_config *gk_conf)
 	int i;
 
 	for (i = 0; i < gk_conf->num_lcores; i++) {
+		unsigned socket = rte_lcore_to_socket_id(gk_conf->lcores[i]);
+		if (ipv6_if_configured(&gk_conf->net->front) &&
+				gk_conf->ipv6_acls[socket] != NULL) {
+			rte_acl_free(gk_conf->ipv6_acls[socket]);
+			gk_conf->ipv6_acls[socket] = NULL;
```

Comment:
Sorry, I should have updated this comment last night, but it ultimately proved unnecessary to use any kind of category. Like you said, the filter should operate like a hardware filter: any filters applied to the interface should be used for all blocks using IPv6 filters on that interface. That's why I was confused during our meeting -- I dropped the idea but forgot about this comment. :)